doc/reference/misc/timeutil.rst

```diff
@@ -0,0 +1,289 @@
+will provide the current time to the resolution of the tick counter
+assuming that the reference and local time instants were precise and
+that the local clock is accurate.
+
+Applications that require precise time should use the other functions in
+this API to provide additional pairs of reference and local time,
+use them to estimate the local clock error, and update the
+transformation with :c:func:`timeutil_realtime_set_skew`.
+
+Note that the timescale produced by :c:func:`timeutil_realtime_get()`
+may not be continuous: non-linear jumps and timestamps earlier than had
+been observed if the underlying alignment state is not carefully
+managed.
+
+.. doxygengroup:: timeutil_realtime_apis
+   :project: Zephyr
+
+.. _timeutil_concepts:
+
+Concepts Underlying Time Support in Zephyr
+******************************************
+
+Terms from `ISO/TC 154/WG 5 N0038 <timeutil-iso8601>`_ (ISO/WD 8601-1),
+paraphrased:
+
+* A *time axis* is a representation of time as instants forming a
+  monotonic (increasing) series.
+* A *time scale* is a way of representing an instant relative to an
+  origin that serves as the epoch.
+* A time scale is *continuous* if the representation map is monotonic and
+  there is exactly one representation for any instant on the underlying
+  time axis.
+
+Relevant Time Scales
+====================
+
+`International Atomic Time <timeutil-TAI>`_ (TAI) is a time scale based
+on averaging clocks that count in SI seconds. TAI is a continuous time
+scale.
+
+`Universal Time <timeutil-UT>`_ (UT) is a time scale based on Earth’s
+rotation. UT is a discontinuous time scale as it requires occasional
+adjustments (`leap seconds <timeutil-wiki-leap-seconds>`_) to maintain
+alignment to changes in Earth’s rotation. Thus the difference between
+TAI and UT varies over time. There are several variants of UT, with `UTC
+<timeutil-UTC>`_ being the most common.
+
+UT times are independent of location. UT is the basis for Standard Time
+(or "local time") which is the time at a particular location. Standard
+time has a fixed offset from UT at any given instant, primarily
+influenced by longitude, but the offset may be adjusted ("daylight
+saving time") to align standard time to the local solar time. In a sense
+local time is "more discontinuous" than UT.
+
+`POSIX Time <timeutil-RFC8536s2>`_ is a time scale that counts seconds since
+the "POSIX epoch" at 1970-01-01T00:00:00Z (i.e. the start of 1970
+UTC). `UNIX Time <timeutil-RFC8536s2>`_ is an extension of POSIX time using
+negative values to represent times before the POSIX epoch. Both of these
+scales assume that every day has exactly 86400 seconds. In normal use
+instants in these scales correspond to times in the UTC scale, so they
+inherit the discontinuity.
+
+The continuous analogue is `UNIX Leap Time <timeutil-RFC8536s2>`_ which
+is UNIX time plus all leap-second corrections added after the POSIX
+epoch (when TAI-UTC was 8 s).
```

Comment:
I don't understand the "when TAI-UTY was 8 s" comment?

Reply:
The Julian date for 1970-01-01 is [2440587.50000](https://www.aavso.org/cgi-bin/cal2jd.pl).

The [formula for TAI-UTC in effect](https://www.iers.org/IERS/EN/Science/Recommendations/resolutionB1.html) on that date is:

    TAI-UTC=   4.2131700 S + (MJD - 39126.) X 0.002592 S

where MJD = JD - 2400000.5.  This evaluates to 8.000082.